pwa/src/templates/componentDetail/ComponentsDetailTemplate.tsx

```diff
@@ -195,13 +200,87 @@ export const ComponentsDetailTemplate: React.FC<ComponentsDetailTemplateProps> =
             <InfoCard
               title=""
               content={
-                <RatingIndicatorTemplate
-                  layoutClassName={styles.ratingIndicatorContainer}
-                  component={_getComponent.data}
-                />
+                <>
+                  {_getComponent.data.embedded?.rating ? (
+                    <RatingIndicatorTemplate
+                      layoutClassName={styles.ratingIndicatorContainer}
+                      maxRating={_getComponent.data.embedded?.rating?.maxRating}
+                      rating={_getComponent.data.embedded?.rating?.rating}
+                    />
+                  ) : (
+                    <div className={styles.noRatingStyle}>{t("No rating available")}</div>
+                  )}
+                  <span
+                    onClick={() => {
+                      show();
+                    }}
+                    className={styles.link}
+                  >
+                    <Link icon={<ArrowRightIcon />} iconAlign="start">
+                      Rating
+                    </Link>
+                  </span>
+                </>
               }
               layoutClassName={styles.infoCard}
             />
+            {isVisible && (
+              <div className={styles.overlay}>
+                <NotificationPopUp
+                  {...{ hide, isVisible }}
+                  title={"Rating"}
+                  description={
+                    <>
+                      {_getComponent.data.embedded?.rating?.rating ? (
+                        <p>{`${_getComponent.data.embedded?.rating?.rating}/${_getComponent.data.embedded?.rating?.maxRating}`}</p>
+                      ) : (
+                        <p>{t("No rating available")}</p>
+                      )}
+
+                      <div className={styles.popupDescription}>
+                        {_getComponent.data.embedded?.rating?.rating >= 1 && (
```

Comment:
You can check if something has a `.length`, if it does, it's an array with items, like so:

```
_getComponent.data.embedded?.rating.rating.length && ( ... )
```